scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -339,6 +339,22 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    */
   def distinct: SCollection[T] = this.pApply(Distinct.create[T]())
 
+  /**
+   * Returns a new SCollection with distinct elements using given function to obtain a
+   * representative value for each input element.
+   *
+   * @param f The funciton to use to get representative values.
+   * @tparam U The type of representative values used to dedup.
+   * @group transform
+   */
+  def distinctWithRepresentativeValueFn[U](f: T => U)(implicit ctu: ClassTag[U])
+  : SCollection[T] = {
+    this.pApply(Distinct
+      .withRepresentativeValueFn(Functions.serializableFn(f.asInstanceOf[T => U]))
```

Comment:
I don't think there's a need for `.asInstanceOf[T => U]` right?

Reply:
You are right, I will remove this.